fsutils/ipcfg/ipcfg.c

```diff
@@ -220,11 +218,14 @@ static int ipcfg_open(FAR const char *netdev, int oflags, mode_t mode)
   if (ret < 0)
     {
       ret = -errno;
+      close(fd);
       fprintf(stderr, "ERROR: Failed to seek to $ld: %d\n",
               (long)CONFIG_IPCFG_OFFSET, ret);
-      close(fd);
+      goto errout_with_path;
     }
+
 #endif
+  ret = fd;
 
```

Comment:
I don't see any logic difference with this change.  What is the problem that you are solving?  Or is this just NIH?

Reply:
Never mind.  It is clearer looking at the whole file.

Reply:
No. It's a bug.
Valid fd is 3 but code returns 0 (OK)
![image](https://user-images.githubusercontent.com/1945821/94732312-f3d83d00-031a-11eb-9453-df9321c1e133.png)